MaxText/layers/attentions.py

```diff
@@ -255,9 +257,67 @@ def apply_attention(
                            Use `dot_product` instead."""
         )
       return self.cudnn_flash_attention(query, key, value, decoder_segment_ids, model_mode), None, None
+    elif self.attention_kernel == "pallas_gpu":
```

Comment:
I think it’s worth rethinking the kernel name, the fact that it uses pallas is an implementation detail. To me, it’s really a `flash` kernel, and then we can pick **which** one based on the config. For ex: ragged or not, autoregressive or not, tpu or gpu.

I propose something like:

```
if use_ragged_attention and model_mode == common_types.MODEL_MODE_AUTOREGRESSIVE:
      if lengths is None:
        lengths = jnp.sum(decoder_segment_ids, axis=-1)

      impl = self.tpu_ragged_attention if config.device_name = ‘tpu’ else self.gpu_ragged_attention

      return impl(query, key, value, lengths, self.ragged_block_size)
```